app/utils/categories.ts

```diff
@@ -42,15 +42,18 @@ export const filterAutoclosedDMs = (
         // Only autoclose DMs that haven't been assigned to a category
         return channelsWithMyChannel;
     }
-
+    const prefMap = preferences.reduce((acc, v) => {
+        const existing = acc.get(v.name);
+        acc.set(v.name, Math.max((v.value as unknown as number) || 0, existing || 0));
+        return acc;
+    }, new Map<string, number>());
     const getLastViewedAt = (cwm: ChannelWithMyChannel) => {
         // The server only ever sets the last_viewed_at to the time of the last post in channel, so we may need
         // to use the preferences added for the previous version of autoclosing DMs.
         const id = cwm.channel.id;
         return Math.max(
             cwm.myChannel.lastViewedAt,
-            getPreferenceValue<number>(preferences, Preferences.CATEGORIES.CHANNEL_APPROXIMATE_VIEW_TIME, id, 0),
-            getPreferenceValue<number>(preferences, Preferences.CATEGORIES.CHANNEL_OPEN_TIME, id, 0),
+            prefMap.get(id) || 0,
```

Comment:
This is where the real issue was. `getPreferenceValue` performs a find over all the preferences, which in this case they are a lot. That is done every channel, and potentially several times.